fontc/src/main.rs

```diff
@@ -233,32 +233,29 @@ struct ChangeDetector {
 }
 
 impl ChangeDetector {
-    fn new(paths: Paths, args: Args) -> Result<(Config, ChangeDetector), Error> {
-        let build_dir = require_dir(paths.build_dir())?;
-        require_dir(paths.glyph_ir_dir())?;
-        let (config, prev_inputs) = init(&build_dir, args).map_err(Error::IoError)?;
-
+    fn new(config: Config, paths: Paths, prev_inputs: Input) -> Result<ChangeDetector, Error> {
         // What sources are we dealing with?
         let mut ir_source = ir_source(&config.args.source)?;
         let current_inputs = ir_source.inputs().map_err(Error::FontIrError)?;
 
-        Ok((
-            config,
-            ChangeDetector {
-                paths,
-                ir_source,
-                prev_inputs,
-                current_inputs,
-            },
-        ))
+        Ok(ChangeDetector {
+            paths,
+            ir_source,
+            prev_inputs,
+            current_inputs,
+        })
     }
 }
 
 fn main() -> Result<(), Error> {
     env_logger::init();
 
     let paths = Paths::new(Path::new("build"));
-    let (config, mut change_detector) = ChangeDetector::new(paths.clone(), Args::parse())?;
+    let build_dir = require_dir(paths.build_dir())?;
+    require_dir(paths.glyph_ir_dir())?;
+    let (config, prev_inputs) = init(&build_dir, Args::parse()).map_err(Error::IoError)?;
```

Comment:
even this `init` function feels a bit double-duty to me. You could make a new config from args and pass it to `init` alongside the build_dir (the latter could also be an attribute if Config btw).

The `init` function could then beme a constructor method for the Input type, that returns an instance of Input given a config